test/jdk/com/sun/jdi/JdwpAllowTest.java

```diff
@@ -76,16 +74,14 @@ public static int handshake(int port) throws IOException {
          return new String[] { jdwpArgs };
     }
 
-    private static Pattern listenRegexp = Pattern.compile("Listening for transport \\b(.+)\\b at address: \\b(\\d+)\\b");
     private static int detectPort(LingeredApp app) {
         long maxWaitTime = System.currentTimeMillis()
                 + Utils.adjustTimeout(10000);  // 10 seconds adjusted for TIMEOUT_FACTOR
         while (true) {
             String s = app.getProcessStdout();
-            Matcher m = listenRegexp.matcher(s);
-            if (m.find()) {
-                // m.group(1) is transport, m.group(2) is port
-                return Integer.parseInt(m.group(2));
+            JDWP.ListenAddress addr = JDWP.parseListenAddress(s);
+            if (addr != null) {
```

Comment:
What is going to happen in the case of addr == null?

Reply:
addr == null means we didn't get "listening for transport" output yet.
We continue to run the loop waiting for the parse-able output